text/0000-types-team.md

```diff
@@ -0,0 +1,240 @@
+
+### Compiler team
+
+* Assessing and implementing fixes for soundness bugs that have to do with the type system.
+* Maintaining a shared libraries that implements the Rust type/trait system and the borrow checker.
+
+### Initiatives
+
+Initiatives that are extending the semantics of the language, such as the [generic associated types](https://rust-lang.github.io/generic-associated-types-initiative/) initiative, will work closely with this team to integrate those changes into the formality models.
+
+## Examples
+
+Here are some examples to illustrate how the types team will interact.
+
+### Evaluating the implication of equality constraints or negative trait bounds
+
+The original RFC for where clauses included equality constraints like `where T == U` (tracked by [#20041]); similarly, people have regularly considered including the option to have "negated" where clauses like `where T: !Debug`. Both of these features, if led to their full generality, turn out to make the implementation of Rust's trait solver significantly more complicated. Therefore, the types team is within its rights to veto these features or to suggest appropriate modifications to how they work and what they mean. On the other hand, it is up to the lang team to decide the syntax of those where clauses and whether they'd be a useful addition to Rust (presuming they had a semantics the types team was happy with). The types team cannot add a feature to Rust all by itself, but it can either remove one (because it is not feasible to implement) or tweak its formal semantics (to ensure it is sound, feasible etc). In this respect, it is just like the compiler team.
+
+[#20041]: https://github.com/rust-lang/rust/issues/20041
+
+### Bug in the type checker
+
+When a soundness bug is found in Rust's type system, the compiler team can contact the types team to request the bug be triaged and to prepare a fix. The types team owns that code and hence is ultimately responsible for reviewing changes to the relevant code.
+
+### Unsafe code guidelines
+
+One initiative worth calling out is the unsafe code guidelines working group. This group is quite old and needs at some point to become active again; at that point, it would effectively be a domain working group, much like the async working group. The role of the types there would be to help integrate and model the unsafe code guidelines proposals and to advise the lang team on the technical implications of what is being proposed. The role of the lang team would be to decide which model end users would prefer and so forth.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Leads
+
+Leads are responsible for:
+
+* Leading and scheduling team meetings
+* Selecting the deep dive meetings
+* Making decisions regarding team membership
+* General "buck stops here"-type decisions
+
+Leads typically serve for 6 months to 1 year, at which point we will consider whether to rotate.
+
+## Team membership
```

Comment:
A common point of confusion for people is whether and how they can request considerationfor becoming a member. We should probably mention that this can be done by contacting the leads?

Reply:
I don't think that's the right approach - asking for membership isn't a great approach. I added a bit that interested people should join meeting or do initiative-specific work.